src/category_theory/subobject/comma.lean

```diff
@@ -0,0 +1,218 @@
+/-
+Copyright (c) 2022 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import category_theory.subobject.well_powered
+
+/-!
+# Subobjects in the category of structured arrows
+
+We compute the subobjects of an object `A` in the category `structured_arrow S T` for `T : C ⥤ D`
+and `S : D` as a subtype of the subobjects of `A.right`. We deduce that `structured_arrow S T` is
+well-powered if `C` is.
+
+## Main declarations
+* `structured_arrow.equiv_subtype`: the order-equivalence between `subobject A` and a subtype of
+  `subobject A.right`.
+
+## Implementation notes
+Our computation requires that `C` has all limits and `T` preserves all limits. Furthermore, we
+require that the morphisms of `C` and `D` are in the same universe. It is possible that both of
+these requirements can be relaxed by refining the results about limits in comma categories.
+
+We also provide the dual results. As usual, we use `subobject (op A)` for the quotient objects of
+`A`.
+
+-/
+
+noncomputable theory
+
+open category_theory.limits opposite
+
+universes v u₁ u₂
+
+namespace category_theory
+variables {C : Type u₁} [category.{v} C] {D : Type u₂} [category.{v} D]
+
+namespace structured_arrow
+variables {S : D} {T : C ⥤ D}
+
+/-- Every subobject of a structured arrow can be projected to a subobject of the underlying
+    object. -/
+def project_subobject [has_limits C] [preserves_limits T] {A : structured_arrow S T} :
+  subobject A → subobject A.right :=
```

Comment:
This is a monotone map right? Is it helpful to define it as such? And then maybe making the order-equiv below might be cleaner?

Reply:
It's true that this is a monotone map, but defining it this way really only moves the proof obligation to a different place, making everything slightly more awkward. For this reason, I would prefer to leave things as they are. This function exists under the same conditions as the full order equiv, so if you need it as a monotone map, you can just use the equiv.